src/github.com/matrix-org/dendrite/mediaapi/writers/download.go

```diff
@@ -167,26 +189,27 @@ func (r *downloadRequest) Validate() *util.JSONResponse {
 	return nil
 }
 
-func (r *downloadRequest) doDownload(w http.ResponseWriter, cfg *config.Dendrite, db *storage.Database, activeRemoteRequests *types.ActiveRemoteRequests, activeThumbnailGeneration *types.ActiveThumbnailGeneration) *util.JSONResponse {
+func (r *downloadRequest) doDownload(
+	w http.ResponseWriter,
+	cfg *config.Dendrite,
+	db *storage.Database,
+	activeRemoteRequests *types.ActiveRemoteRequests,
+	activeThumbnailGeneration *types.ActiveThumbnailGeneration,
+) (*types.MediaMetadata, error) {
 	// check if we have a record of the media in our database
 	mediaMetadata, err := db.GetMediaMetadata(r.MediaMetadata.MediaID, r.MediaMetadata.Origin)
 	if err != nil {
-		r.Logger.WithError(err).Error("Error querying the database.")
-		resErr := jsonerror.InternalServerError()
-		return &resErr
+		return nil, errors.Wrap(err, "Error querying the database.")
```

Comment:
I'd leave out the full stop since this will be formatted as ```"Error querying the database." + ": " + err.Error()```.
Generally people seem to start error messages with lowercase letters so that they scan better in that format.
(I thought we had an automatic linter that checked the error format, but it might just have been kegan.)